File: SwiftDemo/SwiftDemo/CaiNiao/TestArrayClass.swift
//
//  TestArrayClass.swift
//  SwiftDemo
//
//  Created by FlutterRole on 2024/11/21.
//

import Foundation

class TestArrayClass {
    func testArray() {
        testArrayCeate()
        testItem()
        testModify()
        testForIn()
        testMerge()
        testCount()
        testEmpty()
    }
    
    func testArrayCeate() {
        var someArray = [Any]()
        print("someArray1:\(someArray)")
        someArray = [Any](repeating: 0, count: 5)
        print("someArray2:\(someArray)")
        
        var someInts = [Int](repeating: 0, count: 3)
        print("someInts:\(someInts)")
        someInts = [10,20,30]
        print("someInts:\(someInts)")
        
        var someInts1:[Int] = [10,20,30]
        print("someInts1:\(someInts1)")
    }
    
    func testItem() {
        var someInts = [Int](repeating: 10, count: 3)
        var someVar = someInts[0]
        print( "第一个元素的值 \(someVar)" )
        print( "第二个元素的值 \(someInts[1])" )
        print( "第三个元素的值 \(someInts[2])" )
    }
    
    func testModify() {
        var someInts = [Int]()
        someInts.append(contentsOf: 0...1)
        someInts.append(20)
        someInts.append(30)
        someInts += [40]
        someInts[2] = 50
        
        let someVar = someInts[2]
        print( "第一个元素的值 \(someVar)" )
        print( "第二个元素的值 \(someInts[3])" )
        print( "第三个元素的值 \(someInts[4])" )
    }
    
    func testForIn() {
        var someStrs = [String]()
        someStrs.append("Apple")
        someStrs.append("Amazon")
        someStrs.append("Runoob")
        someStrs += ["Google"]
        
        for item in someStrs {
            print(item)
        }
        
        for (index,item) in someStrs.enumerated() {
            print("index is \(index),item is \(item)")
        }
    }
    
    func testMerge(){
        let intsA = [Int](repeating: 2, count:2)
        let intsB = [Int](repeating: 1, count:3)

        var intsC = intsA + intsB

        for item in intsC {
            print(item)
        }
    }
    
    func testCount(){
        let intsA = [Int](repeating: 2, count:2)
        let intsB = [Int](repeating: 1, count:3)
        let intsC = intsA + intsB
        print("intsA count is \(intsA.count)")
        print("intsA count is \(intsB.count)")
        print("intsA count is \(intsC.count)")
    }
    
    func testEmpty(){
        let intsA = [Int](repeating: 2, count:2)
        let intsB = [Int](repeating: 1, count:3)
        var intsC = [Int]()
        print("intsA isEmpty \(intsA.isEmpty)")
        print("intsA isEmpty \(intsB.isEmpty)")
        print("intsA isEmpty \(intsC.isEmpty)")
    }
}

File: flutter_demo/android/app/src/main/kotlin/com/example/flutter_demo/TestArrayClass.kt
package com.example.flutter_demo

class TestArrayClass {
    fun testArray() {
        testCreate()
        testForIn()
        testMerge()
        testCount()
        testEmpty()
    }
    fun testCreate(){
        val a = arrayOf(1,2,3)
        val b = Array(3,{i -> (i*2)})
        println("a is $a, first is $a[0]")
        println("b is $b, first is $b[9]")

        val x:IntArray = intArrayOf(1,2,3)
        x[0] = x[1] + x[2]
        println("x is $x, first is $x[9]")
    }

    fun testForIn() {
        val a = arrayOf(1,2,3)
        for (item in a) {
            println("item is $item")
        }

        for(index in a.indices) {
            println("index is $index, item is ${a[index]}")
        }
    }

    fun testMerge(){
        val a = arrayOf(1,2,3)
        val b = arrayOf(4,5,6)
        val c = a + b
        println("c is $c, first is ${c[4]}")
    }

    fun testCount() {
        val c = arrayOf(1,2,3)
        println("c is $c, count is ${c.count()}")
    }

    fun testEmpty(){
        val a = emptyArray<Int>()
        val c = arrayOf(1,2,3)
        println("a is $a, ${a.isEmpty()}}")
        println("c is $c, ${c.isEmpty()}}")
    }

}